src/apps/filters/h3.c

```diff
@@ -140,35 +185,195 @@ H3Error cellToBoundaryCmd(int argc, char *argv[]) {
     return E_SUCCESS;
 }
 
-bool generalHelp(int argc, char *argv[]) {
-    Arg *args[] = {&helpArg, &cellToLatLngArg, &latLngToCellArg,
-                   &cellToBoundaryArg};
+/// Inspection subcommands
 
-    const char *helpText =
-        "Please use one of the subcommands listed to perform an H3 "
-        "calculation. Use h3 <SUBCOMMAND> --help for details on the usage of "
-        "any subcommand.";
-    return parseArgs(argc, argv, sizeof(args) / sizeof(Arg *), args, &helpArg,
-                     helpText);
+SUBCOMMAND(getResolution, "Extracts the resolution (0 - 15) from the H3 cell") {
+    DEFINE_CELL_ARG(cell, cellArg);
+    Arg *args[] = {&getResolutionArg, &helpArg, &cellArg};
+    PARSE_SUBCOMMAND(argc, argv, args);
+    // TODO: Should there be a general `isValidIndex`?
+    H3Error cellErr = H3_EXPORT(isValidCell)(cell);
+    H3Error edgeErr = H3_EXPORT(isValidDirectedEdge)(cell);
+    H3Error vertErr = H3_EXPORT(isValidVertex)(cell);
+    if (cellErr && edgeErr && vertErr) {
+        return cellErr;
+    }
+    // If we got here, we can use `getResolution` safely, as this is one of the
+    // few functions that doesn't do any error handling (for some reason? I
+    // don't see how this would ever be in a hot loop anywhere.
+    int res = H3_EXPORT(getResolution)(cell);
+    printf("%i", res);
+    return E_SUCCESS;
 }
 
-int main(int argc, char *argv[]) {
-    if (argc <= 1) {
-        printf("Please use h3 --help to see how to use this command.\n");
-        return 1;
+SUBCOMMAND(getBaseCellNumber,
+           "Extracts the base cell number (0 - 121) from the H3 cell") {
+    DEFINE_CELL_ARG(cell, cellArg);
+    Arg *args[] = {&getBaseCellNumberArg, &helpArg, &cellArg};
+    PARSE_SUBCOMMAND(argc, argv, args);
+    // TODO: Should there be a general `isValidIndex`?
+    H3Error cellErr = H3_EXPORT(isValidCell)(cell);
+    H3Error edgeErr = H3_EXPORT(isValidDirectedEdge)(cell);
+    H3Error vertErr = H3_EXPORT(isValidVertex)(cell);
+    if (cellErr && edgeErr && vertErr) {
+        return cellErr;
```

Comment:
I don't object, but for my info, is the principle here that we should always validate CLI input because we assume that correctness is more important than perf in this context?

Reply:
That was my reasoning on this. You have a whole process spawn and arg parse for each call, so the validation logic should be a rounding error here.